### DISCUSSION TITLE
How to pass embedding_exclude_overlap to from_pretrained function ?

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [pyannote/pyannote-audio#1139] [Q&A]
Hi,

I want to test `[embedding_exclude_overlap](https://github.com/pyannote/pyannote-audio/blob/f700d6ea8dedd42e7c822c3b44b46a952e62a585/pyannote/audio/pipelines/speaker_diarization.py#L75)` when doing speaker-diarization to see if it improves the results. I tried to pass it as argument of `from_pretrained`:
```
pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization@2.1",
                                    use_auth_token=ACCESS_TOKEN,
                                    embedding_exclude_overlap=True,
                                    )
``` 
But it does not work.

I also tried `pipeline.embedding_exclude_overlap = True` but it silently discards the change.

So how can I use `embedding_exclude_overlap` ?

Thank you very much for your help!

**Comment by hbredin** [comment]
`pipeline.embedding_exclude_overlap = True` should work just fine.
What makes you think it _"silently discards the change"_?

**Reply by Ca-ressemble-a-du-fake** [reply]
Thank you for your reply. It made me think the changes were discarded because first I did not see any change in the results after switching to true, and second I tried `pipeline.embedding_exclude_overla = True` (note the missing 'p' in the end) and it did not raise any error.

**Reply by hbredin** [reply]
The effect of `embedding_exclude_overlap` is marginal and it is utterly possible that switching it on or off has no impact on the final output for some files.

Also, `embedding_exclude_overlap` [defaults to True in pyannote/speaker-diarization](https://huggingface.co/pyannote/speaker-diarization/blob/main/config.yaml#L7).

**Reply by Ca-ressemble-a-du-fake** [reply]
Good to know thank you (I saw False in code, did not think to look at the diarization pipeline).
